naxsi_src/naxsi_runtime.c

```diff
@@ -886,9 +886,16 @@ ngx_int_t ngx_http_nx_log(ngx_http_request_ctx_t *ctx,
 	strcat(tmp_zone, "FILE_EXT");
       if (mr[i].target_name)
 	strcat(tmp_zone, "|NAME");
+
+      ngx_str_t tmp_val;
+      
+      tmp_val.len = mr[i].name->len + (2 * ngx_escape_uri(NULL, mr[i].name->data, mr[i].name->len, NGX_ESCAPE_URI_COMPONENT));
```

Comment:
This screams "INTEGER OVERFLOW" all over, but since it's in user-supplied configuration, I guess that we don't care that much, do we?